pkg/ipam/types/types.go

```diff
@@ -182,12 +189,18 @@ type IPReleaseStatus string
 //
 // This structure is embedded into v2.CiliumNode
 type IPAMStatus struct {
-	// Used lists all IPs out of Spec.IPAM.Pool which have been allocated
+	// Used lists all IPv4 addresses out of Spec.IPAM.Pool which have been allocated
 	// and are in use.
 	//
 	// +optional
 	Used AllocationMap `json:"used,omitempty"`
 
+	// IPv6Used lists all IPv6 addresses out of Spec.IPAM.IPv6Pool which have been
+	// allocated and are in use.
+	//
+	// +optional
+	IPv6Used AllocationMap `json:"ipv6-used,omitempty"`
+
```

Comment:
In some cases, it makes sense to create a duplicated field that represents IPv6, but in other cases like the one highlighted where the type is a map, do we really need to duplicate the map just to hold IPv6? Couldn't the map hold both IPv4 and IPv6 addresses?

Reply:
@christarazi thank you for the review and feedback. I did consider using the existing `Used` field but I decided to create a separate field for the following reasons:
1. If a user needs to see if an IPv6 address is being used, it will be easier to ignore all the IPs in `Used` and only have to parse IPv6 addresses from `IPv6Used`.
2. The AllocationMap is keyed by a string representing an IP address. If the map supports IPv4 and IPv6 keys then we have to parse and separate IPv4 from IPv6 addresses to perform maintenance for each pool.